### PR TITLE
Show first & last name length errors on front end

### PR DESCRIPTION
**Pull request by Beamanator**
### Details

Idea coming from here: https://github.com/Expensify/App/issues/5091#issuecomment-941681642 - we're starting to standardize showing input errors when submitting the form - validating on the front end instead of on the back-end. This speeds up users seeing errors and prevents some unnecessary API calls.

### Related Issues
Related to: https://github.com/Expensify/App/issues/5091

### Tests
1. Log in with any account

Profile page:
2. Open your profile page (Settings -> Profile)
3. Add a first & last name over 50 characters (ex: `012345678901234567890123456789012345678901234567890`)
4. Try to save the form, verify you see red outlined fields & errors under each
    - Screen Shot 2021-10-20 at 11 35 16 AM
5. Change your first name to be between 0 and 50 characters
6. Resubmit the form, verify you only see the last name field with red outline & error underneath
7. Change your last name to be between 0 and 50 characters
8. Resubmit the form, verify there's no errors and you see the success growl

Request call page:
1. Navigate to your chat with Concierge, click the "Request call" icon
2. Repeat tests from profile page, verifying you see:
    - Errors under both name inputs when their lengths are > 50 characters
    - No errors under either input when their lengths are between 1 and 50 characters
    - An error shows if either the first or last names are empty
    - You still see phone number errors if it's empty or if it's an invalid phone number

### QA Steps
Same as above

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

#### Web

Profile Page:
Screen Shot 2021-10-20 at 11 35 16 AM

Request a call:

https://user-images.githubusercontent.com/3885503/138144719-95599f85-c8ff-4b3d-a02d-53935a59371f.mov

#### Mobile Web

Profile Page:
Screen Shot 2021-10-20 at 12 20 39 PM

Request a call:

https://user-images.githubusercontent.com/3885503/138149437-f5ef686e-4fce-4325-acaa-84926381b17e.mov

#### Desktop

Profile Page:
Screen Shot 2021-10-20 at 11 57 31 AM

Request a call:

https://user-images.githubusercontent.com/3885503/138147459-72d0d0d7-e567-4413-9f91-10f947feca54.mov

#### iOS

Profile Page:
Screen Shot 2021-10-20 at 12 16 56 PM

Request a call:

https://user-images.githubusercontent.com/3885503/138149020-27eba4a2-b055-456a-861e-82206a9cfb29.mov

#### Android

Profile Page:
Screen Shot 2021-10-20 at 12 04 20 PM

Request a call:

https://user-images.githubusercontent.com/3885503/138147987-5ee753a9-2fba-4bfc-a94b-c781ff5a269e.mov

### CODE REVIEW

File: src/libs/ValidationUtils.js [hunk 1]
```diff
@@ -230,6 +230,15 @@ function isNumericWithSpecialChars(input) {
     return /^\+?\d*$/.test(input.replace(/[()-]/g, ''));
 }
 
+/**
+ * Checks weather a given first or last name is valid
```
Comment (nickmurray47) [inline src/libs/ValidationUtils.js:234]: nab: 
```suggestion
 * Checks whether a given first or last name is valid
```

File: src/libs/ValidationUtils.js [hunk 2]
```diff
@@ -230,6 +230,15 @@ function isNumericWithSpecialChars(input) {
...
+ * @param {String} name
+ * @returns {Boolean}
+ */
+function isValidFirstOrLastName(name) {
```
Comment (nickmurray47) [inline src/libs/ValidationUtils.js:238]: also nab I think: would prefer `isValidLengthForFirstOrLastName` as a more descriptive name. I feel like given the current method name I'd assume we'd also check for invalid characters, etc.

File: src/pages/RequestCallPage.js [hunk 3]
```diff
@@ -162,13 +171,30 @@ class RequestCallPage extends Component {
...
+     */
+    validateInputs() {
+        const firstOrLastNameEmpty = _.isEmpty(this.state.firstName.trim()) || _.isEmpty(this.state.lastName.trim());
+        if (firstOrLastNameEmpty) {
+            Growl.error(this.props.translate('requestCallPage.growlMessageEmptyName'));
         }
+
+        this.setState((prevState) => {
+            const phoneNumberError = this.getPhoneNumberError(prevState.phoneNumber);
+            const nameErrors = getFirstAndLastNameErrors(prevState.firstName, prevState.lastName);
```
Comment (nickmurray47) [inline src/pages/RequestCallPage.js:189]: Looks like we just need to initialize these outside the `setState` call

**Review by nickmurray47 — CHANGES_REQUESTED** [review]
A few small linting errors, but the rest lgtm

File: src/pages/RequestCallPage.js (hunk 3 above)
Comment (Beamanator) [reply src/pages/RequestCallPage.js:189]: wait i may be confused lol

Comment (Beamanator) [reply src/pages/RequestCallPage.js:189]: No I don't think I'm confused anymore. `setState` is async so I'm trying to figure out how to get the timing right haha

Comment (Beamanator) [reply src/pages/RequestCallPage.js]: Huh, turns out a bit of destructuring cured or confused eslint haha

**Comment by Beamanator** [comment]
Tests passed! 💪

File: src/libs/actions/PersonalDetails.js [hunk 4]
```diff
@@ -64,6 +65,21 @@ function getDisplayName(login, personalDetail) {
...
+ * those errors get returned as empty strings.
+ *
+ * @param {String} firstName
+ * @param {String} lastName
+ * @returns {Object}
+ */
+function getFirstAndLastNameErrors(firstName, lastName) {
+    return {
+        firstName: isValidLengthForFirstOrLastName(firstName) ? '' : translateLocal('personalDetails.error.firstNameLength'),
+        lastName: isValidLengthForFirstOrLastName(lastName) ? '' : translateLocal('personalDetails.error.lastNameLength'),
```
Comment (nickmurray47) [inline src/libs/actions/PersonalDetails.js:79]: Last comment for clarity here and when we use the errors in the `setState` calls below
```suggestion
        firstNameError: isValidLengthForFirstOrLastName(firstName) ? '' : translateLocal('personalDetails.error.firstNameLength'),
        lastNameError: isValidLengthForFirstOrLastName(lastName) ? '' : translateLocal('personalDetails.error.lastNameLength'),
```

**Review by nickmurray47 — COMMENTED** [review]
Looks good - had one more small comment for clarity!

File: src/libs/actions/PersonalDetails.js (hunk 4 above)
Comment (Beamanator) [reply src/libs/actions/PersonalDetails.js:79]: Yeah I was thinking about that and I don't know why I didn't make this change. Definitely clearer this way 👍

**Comment by Beamanator** [comment]
Updated! 👍

**Review by nickmurray47 — APPROVED** [review]
👍 nice!

**Pull request merged**